### PR TITLE
Use the correct replacement classes. 

### PR DESCRIPTION
**Pull request by vRallev**
This is a regression that was introduced while working on the new CodeGenerator API.

### CODE REVIEW

File: compiler/src/main/java/com/squareup/anvil/compiler/codegen/BindingModuleGenerator.kt [hunk 1]
```diff
@@ -190,6 +190,12 @@ internal class BindingModuleGenerator(
               annotation = contributesToFqName,
               scope = scope
             )
+            .filter { it.annotationOrNull(daggerModuleFqName) != null }
+            .flatMap {
+              it.annotationOrNull(contributesToFqName, scope)
+                ?.replaces(module)
+                ?: emptyList()
+            }
```
Comment (vRallev) [inline compiler/src/main/java/com/squareup/anvil/compiler/codegen/BindingModuleGenerator.kt:198]: I simply forgot to add these lines back. I tried to write a test for that, but it's tricky. It requires to load descriptors from precompiled artifacts. I could do this in the integration tests, but it's kinda awkward to test the edge case there.

**Review by ZacSweers — APPROVED** [review]
